graphql2/graphqlapp/calendarsubscription.go

```diff
@@ -0,0 +1,100 @@
+package graphqlapp
+
+import (
+	"context"
+	"database/sql"
+	"github.com/target/goalert/calendarsubscription"
+	"github.com/target/goalert/graphql2"
+	"github.com/target/goalert/search"
+)
+
+func (q *Query) CalendarSubscription(ctx context.Context, id string) (*calendarsubscription.CalendarSubscription, error) {
+	return q.CalendarSubscriptionStore.FindOne(ctx, id)
+}
+
+// todo: return url instead of bool once endpoint has been created
+func (m *Mutation) CreateCalendarSubscription(ctx context.Context, input graphql2.CreateCalendarSubscriptionInput) (res bool, err error) {
+	err = withContextTx(ctx, m.DB, func(ctx context.Context, tx *sql.Tx) error {
+		cs := &calendarsubscription.CalendarSubscription{
+			Name: input.Name,
+		}
+
+		res, err = m.CalendarSubscriptionStore.CreateSubscriptionTx(ctx, tx, cs)
+		if err != nil {
+			return err
+		}
+
+		// todo: gen url for user
+
+		return err
+	})
+
+	return res, err
+}
+
+func (m *Mutation) UpdateCalendarSubscription(ctx context.Context, input graphql2.UpdateCalendarSubscriptionInput) (bool, error) {
+	err := withContextTx(ctx, m.DB, func(ctx context.Context, tx *sql.Tx) error {
+		cs := &calendarsubscription.CalendarSubscription{
```

Comment:
This needs some sort of `FindOneForUpdateTx` and only update the provided fields.

For example, you should be able to update the name without providing the `disabled` value.

See `UpdateUserContactMethod` for an example: https://github.com/target/goalert/blob/master/graphql2/graphqlapp/contactmethod.go#L76